### DISCUSSION TITLE
Stratux AHRS

### DISCUSSION THREAD

**Discussion by e100** [makerplane/FIX-Gateway#168] [General]
@EdB987 

Thought the conversation would be better in a discussion and will try to answer all the questions you had.
My installation, and more specifically the installation steps for it, are documented here:
https://github.com/e100/makerplane-configs/blob/main/INSTALL.md

All of that is specific to how I setup my systems, but should help you with ideas and solving problems.

You can use Mavlink as an AHRS/GPS input without using it for AP/Trim.
With that, I have written a hacked patch for Stratux so it will get the AHRS data from the Fix Gateway.
https://github.com/e100/makerplane-configs/blob/main/stratux/README.md
I do not really like this hack but it does work, would like to come up with something more elegant and less invasive to Stratux.

I agree, the Stratux AHRS does not work well, one of the reasons I do not use it.

The x729 RTC issue conflicting with the RPi RTC:
https://github.com/e100/makerplane-configs/blob/main/INSTALL.md#optional---setup-rtc-device-on-the-x729

You had mentioned having issues with wlan0 and ap0 not coming up on boot, All I have done to make that work is what I've documented in my makerplane-configs repository. I think this might be the specific changes needed to make it work automatically:
https://github.com/e100/makerplane-configs/blob/main/stratux/README.md#modify-network-template
But it is possible some other part of my setup is also important.

**Comment by EdB987** [comment]
Thanks for the responses! For the RTC issue, I resolved the conflict by enabling the 2nd i2c bus and moving the Stratux AHRS there. But now that I've decided to follow your lead on Mavlink - don't need it any more.

One other question is suitable sunlight readable displays. I have a Makerplane 5" display - but it's going to be too small if I go down the iFly path. The setup I tested this weekend is using my iPad mini - perfect size, but very hard to see in bright sunshine. My Canopy is a bubble with no shade at the moment but will install some soon. Are your current displays bright enough? 

For the Ap0 and wlan0 issues - I dunno - I resolved them somehow - seem to be working now.

**Comment by EdB987** [comment]
Additional: yours and Phil's efforts are spectacular and very much appreciated!

**Comment by EdB987** [comment]
Eric - need some advice...
Things are coming along - the airplane is flying and I've been able to configure most of my sensors. I have an RDAC XF with MAP but am seeing low TC readings, even with high quality sensors. I see about 190 F when I immerse a CHT TC in boiling water and I see the RDAC reports ambient temp of 62F when the real ambient temp is 73F.  This leads me to believe I have an RDAC calibration issue with the RDAC onboard sensor reading too low and needing adjustment up about 5 degrees C to bring everything inline.  Would you agree with that?

I have tried mightily to get the RDAC to accept new calibration but I have not discovered the magic of ho to do that yet. I'll end up calling MGL to hear what they have to say,

Alternatively, I thought about just adding calibration mappings in the mgl.yaml file - but that seems a bit hokey, but might work for now. 

Other items in my setup that may be of interest: I'm using an iPad Pro as my screen connected via NoMachine X client - this works surprisingly well except I've spent a huge amount of time trying to force my Pi5 to pretend to have a display connected. With a real display connected, everything just works. When it's headless, it's much more finicky - I've had to settle for 1024x768 resolution for now. Not too great.

The FDR function is fantastic and I use it a lot while I'm working through my temp issues. I have a set of steam gauges in addition to makerplane for peace of mind.

Any ideas on the calibration issues?

**Comment by e100** [comment]
In my MiniMax I have some MGL Blaze displays that show data from the RDAC. The RDAC itself does not do any calibration that I am aware of. It is done in the display. In the MGL blaze you can select some preset calibrations or do a custom calibration.  The rdac.yaml calibration was inspired by how it's done in the MGL display where you select how many calibration points you want and set the desired display value vs RDAC output value. Is your RDAC ground wire connected directly to your engine with nothing else but the RDAC on that ground wire? If not, try doing that. I was chasing high temps until I did that. Any current draw across the ground between the RDAC and engine will cause thermocouple senor reading issues.
If the displayed temp value changes when you turn on and off electrical devices, you have a ground issue. However if no change is observed you could still have a ground issue. 
I calibrated my engine coolant temp by placing the sensor and a good thermometer in boiling water. When I turned the heat off I wrote down the rdac output every 10 degrees as it cooled down then programmed the display or if using pyefis update the rdac.yaml.
